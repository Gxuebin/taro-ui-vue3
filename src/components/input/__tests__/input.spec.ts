import { mountFactory } from '@/tests/helper'
import AtInput from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mountFactory(AtInput, {}, values, slots)
}

describe('AtInput', () => {
  it('should render default AtInput', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'border:none;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- value', () => {
    const wrapper = factory({ value: 'value' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- name', () => {
    const wrapper = factory({ name: 'name' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- placeholder', () => {
    const wrapper = factory({ placeholder: 'placeholder' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- title', () => {
    const wrapper = factory({ title: 'title' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- maxLength', () => {
    const wrapper = factory({ maxLength: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(number)', () => {
    const wrapper = factory({ type: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(password)', () => {
    const wrapper = factory({ type: 'password' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(idcard)', () => {
    const wrapper = factory({ type: 'idcard' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(digit)', () => {
    const wrapper = factory({ type: 'digit' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- type(phone)', () => {
    const wrapper = factory({ type: 'phone' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- border', () => {
    const wrapper = factory({ border: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- editable', () => {
    const wrapper = factory({ editable: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- error', () => {
    const wrapper = factory({ error: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- clear', () => {
    const wrapper = factory({ clear: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- required', () => {
    const wrapper = factory({ required: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtInput Behavior', () => {
  it('should trigger onChange', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange })
    wrapper.find('.at-input__input').trigger('input', { detail: { value: 'value' } })
    expect(onChange).toBeCalled()
  })

  it('should trigger onFocus', () => {
    const onFocus = jest.fn()
    const wrapper = factory({ onFocus: onFocus })
    wrapper.find('.at-input__input').trigger('focus', { height: 30 })
    expect(onFocus).toBeCalled()
  })

  it('should trigger onConfirm', () => {
    const onConfirm = jest.fn()
    const wrapper = factory({ onConfirm: onConfirm })
    wrapper.find('.at-input__input').trigger('confirm', { detail: { value: 'value' } })
    expect(onConfirm).toBeCalled()
  })

  it('should trigger onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, editable: false })
    wrapper.find('.at-input .at-input__overlay').trigger('tap')
    expect(onClick).toBeCalled()
  })

  it('should trigger onChange when prop clear is true', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange, clear: true, value: 'value' })
    wrapper.find('.at-input .at-input__icon').trigger('touchstart')
    expect(onChange).toBeCalled()
  })

  it('should trigger onErrorClick when error is true', () => {
    const onErrorClick = jest.fn()
    const wrapper = factory({ onErrorClick: onErrorClick, error: true, value: 'value' })
    wrapper.find('.at-input .at-input__icon').trigger('touchstart')
    expect(onErrorClick).toBeCalled()
  })
})
