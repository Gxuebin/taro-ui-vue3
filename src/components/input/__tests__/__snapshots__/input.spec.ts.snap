// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtInput should render default AtInput 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- border 1`] = `
<view
  class="at-input"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- clear 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- disabled 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--disabled"
  >
    <view
      class="at-input__overlay"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- editable 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--disabled"
  >
    <view
      class="at-input__overlay"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- error 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--error"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <view
      class="at-input__icon"
    >
      <text
        class="at-icon at-icon-alert-circle at-input__icon-alert"
      />
    </view>
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- maxLength 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="10"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- name 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name="name"
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- placeholder 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder="placeholder"
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- required 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- style 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- title 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <label
      class="at-input__title"
      for=""
    >
      title
    </label>
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- type(digit) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="digit"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- type(idcard) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="idcard"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- type(number) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="number"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- type(password) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="true"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- type(phone) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="11"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="number"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;

exports[`AtInput should render prop -- value 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!---->
    <input
      adjustposition="false"
      autofocus="false"
      class="at-input__input"
      confirmtype="done"
      cursor="0"
      cursorspacing="50"
      id="weui-input_2020"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholderclass="placeholder "
      placeholderstyle=""
      type="text"
    />
    <!---->
    <!---->
    <view
      class="at-input__children"
    />
  </view>
</view>
`;
